### PR TITLE
Benchmarks: Add benchmark -  add micro benchmark for cublas test

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
Add benchmark, example, test for cublas test

### CODE REVIEW

File: superbench/benchmarks/micro_benchmarks/cublas_para_info.json [hunk 1]
```diff
@@ -0,0 +1,175 @@
+[
```
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/cublas_para_info.json:1]: @abuccts how can we handle it?

File: superbench/benchmarks/micro_benchmarks/__init__.py [hunk 2]
```diff
@@ -7,7 +7,9 @@
 from superbench.benchmarks.micro_benchmarks.sharding_matmul import ShardingMatmul
 from superbench.benchmarks.micro_benchmarks.computation_communication_overlap import ComputationCommunicationOverlap
 from superbench.benchmarks.micro_benchmarks.kernel_launch_overhead import KernelLaunch
+from superbench.benchmarks.micro_benchmarks.cublas_test import CublasFunction
 
 __all__ = [
-    'MicroBenchmark', 'MicroBenchmarkWithInvoke', 'ShardingMatmul', 'ComputationCommunicationOverlap', 'KernelLaunch'
+    'MicroBenchmark', 'MicroBenchmarkWithInvoke', 'ShardingMatmul', 'ComputationCommunicationOverlap', 'KernelLaunch',
+    'CublasFunction'
```
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/__init__.py:14]: CublasBenchmark?

File: examples/benchmarks/cublas_test.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+Commands to run:
+  python3 examples/benchmarks/cublas_test.py
+"""
+
+from superbench.benchmarks import BenchmarkRegistry
+from superbench.common.utils import logger
+
+if __name__ == '__main__':
+    parameters = '--num_warmup 8 --num_steps 100 --num_in_step 1000 --random_seed 1'
```
Comment (cp5555) [inline examples/benchmarks/cublas_test.py:14]: Can we change the random seed to 33931?

File: superbench/benchmarks/micro_benchmarks/cublas_test.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+import os
+import json
+from pathlib import Path
+
+from superbench.common.utils import logger
+from superbench.benchmarks import BenchmarkRegistry
+from superbench.benchmarks.micro_benchmarks import MicroBenchmarkWithInvoke
+
+
+class CublasFunction(MicroBenchmarkWithInvoke):
```
Comment (cp5555) [inline superbench/benchmarks/micro_benchmarks/cublas_test.py:15]: CublasBenchmark？

File: superbench/benchmarks/micro_benchmarks/cublas_test.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+            '--num_in_step',
+            type=int,
+            default=1000,
+            required=False,
+            help='The number of functions in one step.',
+        )
+        self._parser.add_argument(
+            '--random_seed',
+            type=int,
+            default=0,
```
Comment (abuccts) [inline superbench/benchmarks/micro_benchmarks/cublas_test.py:55]: defult to a non-zero seed?

File: superbench/benchmarks/micro_benchmarks/cublas_test.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+        command += (' --num_test ' + str(self._args.num_steps))
+        command += (' --warm_up ' + str(self._args.num_warmup))
+        command += (' --num_in_step ' + str(self._args.num_in_step))
+        command += (' --random_seed ' + str(self._args.random_seed))
+        try:
+            with open(self._args.config_path, 'r') as load_config:
+                config_array = json.load(load_config)
+            for config_json in config_array:
+                config_json_str = "\'" + json.dumps(config_json).replace(' ', '') + "\'"
+                complete_command = command + (' --config_json ' + str(config_json_str))
```
Comment (abuccts) [inline superbench/benchmarks/micro_benchmarks/cublas_test.py:86]: why need `str(config_json_str)`?

Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/cublas_test.py]: removed

File: superbench/benchmarks/micro_benchmarks/__init__.py (hunk 2 above)
Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/__init__.py]: renamed

File: superbench/benchmarks/micro_benchmarks/cublas_test.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+            '--random_seed',
+            type=int,
+            default=0,
+            required=False,
+            help='The random seed to fill in the data of the function.',
+        )
+        self._parser.add_argument(
+            '--config_path',
+            type=str,
+            default=Path(__file__).parent / 'cublas_para_info.json',
```
[the remark this answers is not in the document]
Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/cublas_test.py]: thank you for the comment,  added

File: superbench/benchmarks/micro_benchmarks/cublas_test.py (hunk 5 above)
Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/cublas_test.py]: default to  33931

File: examples/benchmarks/cublas_test.py (hunk 3 above)
Comment (yukirora) [reply examples/benchmarks/cublas_test.py]: set as default and remove here

File: superbench/benchmarks/micro_benchmarks/cublas_para_info.json (hunk 1 above)
Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/cublas_para_info.json]: convert json file to built-in dict list in init of the benchmark，and add a interface for user to run custom configuration with json/yaml str

File: examples/benchmarks/cublas_test.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""Micro benchmark example for matmul with pytorch.
```
Comment (guoshzhao) [inline examples/benchmarks/cublas_test.py:4]: update the comments.

File: superbench/benchmarks/micro_benchmarks/cublas_test.py [hunk 9]
```diff
@@ -0,0 +1,307 @@
...
+import json
+import yaml
+
+from superbench.common.utils import logger
+from superbench.benchmarks import BenchmarkRegistry
+from superbench.benchmarks.micro_benchmarks import MicroBenchmarkWithInvoke
+
+
+class CublasBenchmark(MicroBenchmarkWithInvoke):
+    """The CublasBenchmark overhead benchmark class."""
```
Comment (guoshzhao) [inline superbench/benchmarks/micro_benchmarks/cublas_test.py:16]: not overhead, better use 'Cublas performance benchmark'

File: superbench/benchmarks/micro_benchmarks/cublas_test.py [hunk 10]
```diff
@@ -0,0 +1,307 @@
...
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+
+        self.default_params_dict_list = [
```
Comment (guoshzhao) [inline superbench/benchmarks/micro_benchmarks/cublas_test.py:26]: private member, can use self.__default_params_dict_list

File: superbench/benchmarks/micro_benchmarks/cublas_test.py [hunk 11]
```diff
@@ -0,0 +1,307 @@
...
+            cmd_idx (int): the index of command corresponding with the raw_output.
+            raw_output (str): raw output string of the micro-benchmark.
+
+        Return:
+            True if the raw output string is valid and result can be extracted.
+        """
+        self._result.add_raw_data('raw_output_' + str(cmd_idx), raw_output)
+
+        try:
+            lines = raw_output.split('\n')
```
Comment (guoshzhao) [inline superbench/benchmarks/micro_benchmarks/cublas_test.py:273]: Use `splitlines()` directly?

File: superbench/benchmarks/micro_benchmarks/cublas_test.py [hunk 12]
```diff
@@ -0,0 +1,307 @@
...
+            logger.error(
+                'Error in running cublas test - round: {}, index of cmd: {}, benchmark: {}, raw data: {}'.format(
+                    self._curr_run_index, cmd_idx, self._name, raw_output
+                )
+            )
+            return False
+        return True
+
+
+BenchmarkRegistry.register_benchmark('cublas-test', CublasBenchmark)
```
Comment (guoshzhao) [inline superbench/benchmarks/micro_benchmarks/cublas_test.py:307]: Specify the platform, refer:
`BenchmarkRegistry.register_benchmark('cublas-test', CublasBenchmark, platform=Platform.CUDA)`

File: superbench/benchmarks/micro_benchmarks/cublas_test.py (hunk 10 above)
Comment (cp5555) [reply superbench/benchmarks/micro_benchmarks/cublas_test.py:26]: I agree.

**Review by cp5555 — APPROVED** [review]

**Review by abuccts — APPROVED** [review]

File: examples/benchmarks/cublas_test.py (hunk 8 above)
Comment (yukirora) [reply examples/benchmarks/cublas_test.py]: sure, thank you

File: superbench/benchmarks/micro_benchmarks/cublas_test.py (hunk 11 above)
Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/cublas_test.py]: ok

File: superbench/benchmarks/micro_benchmarks/cublas_test.py (hunk 10 above)
Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/cublas_test.py]: ok, thank you for the comment

File: superbench/benchmarks/micro_benchmarks/cublas_test.py (hunk 12 above)
Comment (yukirora) [reply superbench/benchmarks/micro_benchmarks/cublas_test.py]: ok, thank you for the comment

**Pull request merged**